drivers/ethernet/eth_e1000.c

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "ETH E1000"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETHERNET_LEVEL
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <net/ethernet.h>
+#include <pci/pci.h>
+#include "eth_e1000_priv.h"
+
+static const char *e1000_reg_to_string(enum e1000_reg_t r)
+{
+#define _(_x)	case _x: return #_x
+	switch (r) {
+	_(CTRL);
+	_(ICR);
+	_(ICS);
+	_(IMS);
+	_(RCTL);
+	_(TCTL);
+	_(RDBAL);
+	_(RDBAH);
+	_(RDLEN);
+	_(RDH);
+	_(RDT);
+	_(TDBAL);
+	_(TDBAH);
+	_(TDLEN);
+	_(TDH);
+	_(TDT);
+	_(RAL);
+	_(RAH);
+	}
+	E("Unsupported register: 0x%x", r);
+	k_oops();
+	return NULL;
+}
+
+static enum ethernet_hw_caps e1000_caps(struct device *dev)
+{
+	return  ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T | \
+		ETHERNET_LINK_1000BASE_T;
+}
+
+static size_t e1000_linearize(struct net_pkt *pkt, void *buf, size_t bufsize)
```

Comment:
Are you sure that we *still* don't have a reusable function which does that? ;-)